src/core/data-structures/finite-state-machine.js

```diff
@@ -65,11 +65,7 @@ export class FiniteStateMachine {
     this.state_ = newState;
 
     const transition = this.statesToTransition_(oldState, newState);
-    const callback = this.transitions_[transition];
-
-    if (callback) {
-      callback();
-    }
+    this.transitions_[transition]?.();
```

Comment:
This actually exposes the `transitions_` map as the `this` context. Splitting into `callback?.()` will prevent that.

Reply:
Oh I see, that makes sense. Fixed.